src/main/scala/no/gdl/bookapi/service/search/IndexService.scala

```diff
@@ -0,0 +1,296 @@
+        intField("externalId"),
+        keywordField("uuid") index false,
+        languageField("title", language),
+        languageField("description", language),
+        objectField("translatedFrom").fields(
+          keywordField("code"),
+          keywordField("name")
+        ),
+        objectField("language").fields(
+          keywordField("code"),
+          keywordField("name")
+        ),
+        nestedField("availableLanguages").fields(
+          keywordField("code"),
+          keywordField("name")
+        ),
+        objectField("licence").fields(
+          intField("id"),
+          intField("revision"),
+          keywordField("name"),
+          keywordField("description"),
+          keywordField("url")
+        ),
+        objectField("publisher").fields(
+          intField("id"),
+          intField("revision"),
+          keywordField("name")
+        ),
+        keywordField("readingLevel"),
+        keywordField("typicalAgeRange"),
+        keywordField("educationalUse"),
+        keywordField("educationalRole"),
+        keywordField("timeRequired"),
+        dateField("datePublished"),
+        dateField("dateCreated"),
+        dateField("dateArrived"),
+        nestedField("categories").fields(
+          textField("category")
+        ),
+        objectField("coverPhoto").fields(
+          textField("large"),
+          textField("small")
+        ),
+        objectField("downloads").fields(
+          textField("epub"),
+          textField("pdf")
+        ),
+        //nestedField("tags"),
+        nestedField("contributors").fields(
+          intField("id"),
+          intField("revision"),
+          keywordField("type"),
+          textField("name")
+        ),
+        nestedField("chapters").fields(
+          intField("id"),
+          intField("seqNo"),
+          languageField("title", language),
+          textField("url")
+        ),
+        booleanField("supportsTranslation")
+      ))
+    }
+
+    private def languageField(fieldName: String, languageTag: LanguageTag) = {
+      val languageAnalyzer = findByLanguage(Some(languageTag.language.id))
+      val languageSupportedField = TextFieldDefinition(fieldName).fielddata(true) analyzer languageAnalyzer.get.analyzer
+      languageSupportedField
+    }
+
+    def findAllIndexes(): Try[Seq[String]] = {
+      esClient.execute(
+        getAliases()
+      ) match {
+        case Failure(failure) => errorHandler(Failure(failure))
+        case Success(response) => Success(response.result.mappings.keys.toSeq.map(_.name))
+      }
+    }
+
+    def updateAliasTarget(oldIndexName: Option[String], newIndexName: String, languageTag: LanguageTag): Try[Any] = {
+      if (!indexExisting(newIndexName).getOrElse(false)) {
+        Failure(new IllegalArgumentException(s"No such index: $newIndexName"))
+      } else {
+        var actions = List[AliasActionDefinition](AddAliasActionDefinition(BookApiProperties.searchIndex(languageTag), newIndexName))
```

Comment:
Samme comment about for comprehension instead of mutation.

Reply:
Agree, but RemoveAliasAction can only be present if the old index exists. Any propositions to how to accomplish this in scala?